nixos/modules/services/misc/mollysocket.nix

```diff
@@ -0,0 +1,105 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.mollysocket;
+  configuration = format.generate "mollysocket.conf" cfg.settings;
+  format = pkgs.formats.toml { };
+in {
+  options.services.mollysocket = {
+    enable = mkEnableOption (mdDoc ''
+      [MollySocket](https://github.com/mollyim/mollysocket) for getting Signal
+      notifications via UnifiedPush
+    '');
+
+    settings = mkOption {
+      default = { };
+      description = mdDoc ''
+        Configuration for MollySocket. Available options are listed
+        [here](https://github.com/mollyim/mollysocket#configuration).
+      '';
+      type = types.submodule {
+        freeformType = format.type;
+        options = {
+          host = mkOption {
+            default = "127.0.0.1";
+            description = mdDoc "Listening address of the web server";
+            type = types.str;
+          };
+
+          port = mkOption {
+            default = 8020;
+            description = mdDoc "Listening port of the web server";
+            type = types.port;
+          };
```

Comment:
These are not required if MollySocket is run in Air Gapped mode, see https://github.com/mollyim/mollysocket?tab=readme-ov-file#web-server